### PR TITLE
Fixed -> DropdownMenu throws exception when it is in any scrollable l…

### PR DESCRIPTION
**Pull request by faisalansari0367**
Fixed -> DropdownMenu throws exception when it is in any scrollable list view and scrolls quickly #139871

### CODE REVIEW

**Comment by faisalansari0367** [comment]
I'd appreciate some assistance in writing the necessary tests to cover the changes introduced in this PR.

Could someone with expertise in testing or familiarity with this part of the codebase kindly review and contribute by creating the required tests? Your input will significantly enhance the reliability and robustness of the code.

Thank you in advance for your collaboration!

**Comment by navaronbracke** [comment]
You could write a widget test that does the following:

- pump a ListView.builder with the WidgetTester; this widget contains a lot of Containers and a dropdown somewhere in the middle
- start a fling gesture with a high enough velocity that scrolls the list fast enough to both create the Dropdown widget state and  to make the widget unmount before the post frame callback ran

**Comment by faisalansari0367** [comment]
> You could write a widget test that does the following:
> 
> * pump a ListView.builder with the WidgetTester; this widget contains a lot of Containers and a dropdown somewhere in the middle
> * start a fling gesture with a high enough velocity that scrolls the list fast enough to both create the Dropdown widget state and  to make the widget unmount before the post frame callback ran
> * dispose of the gesture that you created, at the end of the test
> * add an `expect(tester.takeException(), isNull);` to verify that no exception happened

Thanks @navaronbracke 

I have added the tests for the same.

File: packages/flutter/test/material/dropdown_menu_test.dart [hunk 1]
```diff
@@ -1907,6 +1907,37 @@ void main() {
     expect(selectionCount, 1);
     expect(tester.takeException(), isNull);
   });
+
+
+  // Test for #139871
+  // DropdownMenu throws exception when it is in any scrollable list view and scrolls quickly #139871
+  // https://github.com/flutter/flutter/issues/139871
```
Comment (goderbauer) [inline packages/flutter/test/material/dropdown_menu_test.dart:1914]: fly-by nit: convention is to just say:

```suggestion
  // Regression test for https://github.com/flutter/flutter/issues/139871.
```

**Review by QuncCccccc — COMMENTED** [review]
Thanks a lot for the contribution! This looks good:) Could you help rebase master to fix the conflicts?

**Review by QuncCccccc — APPROVED** [review]
LGTM:) Thanks for the contribution!

**Review by MitchellGoodwin — APPROVED** [review]
LGTM! Thank you for putting this together.

**Pull request merged**